patches/tModLoader/Terraria/Projectile.TML.cs

```diff
@@ -17,15 +17,75 @@ public partial class Projectile : IEntityWithGlobals<GlobalProjectile>
 		public static Projectile NewProjectileDirect(IEntitySource spawnSource, Vector2 position, Vector2 velocity, int type, int damage, float knockback, int owner = 255, float ai0 = 0f, float ai1 = 0f)
 			=> Main.projectile[NewProjectile(spawnSource, position.X, position.Y, velocity.X, velocity.Y, type, damage, knockback, owner, ai0, ai1)];
 
-		private DamageClass _damageClass = DamageClass.Generic;
+		private DamageClass _damageClass = DamageClass.Default;
 		/// <summary>
-		/// The damage type of this Projectile. Assign to DamageClass.Generic/Melee/Ranged/Magic/Summon/Throwing, or ModContent.GetInstance&lt;T&gt;() for custom damage types.
+		/// The damage type assigned to this projectile, represented as a DamageClass.
+		/// Leave blank or use DamageClass.Default to prevent damage type scaling of any kind for this projectile.
+		/// Use DamageClass.Generic/Melee/Ranged/Magic/Summon/Throwing for vanilla damage types.
+		/// Refer to ExampleMod for more information on how to create and use your own damage types.
 		/// </summary>
 		public DamageClass DamageType {
 			get => _damageClass;
 			set => _damageClass = value ?? throw new ArgumentException("DamageType cannot be null");
 		}
 
+		private int _armorPenetration = 0;
+		/// <summary>
+		/// The number of defense points that this projectile can ignore on its own. Cannot be set to negative values. Defaults to 0.
+		/// On spawn, if this projectile was fired from a weapon, this value has the total armor penetration of the weapon that made the projectile added to itself.
+		/// </summary>
+		public int ArmorPenetration {
+			get => _armorPenetration;
+			set {
+				if (value < 0)
+					throw new Exception("A projectile's armor penetration value cannot be set below 0.");
+				else
+					_armorPenetration = value;
```

Comment:
```suggestion
				if (value < 0) throw new ArgumentException($"{nameof(Projectile)}.{nameof(ArmorPenetration)} must be >= 0");
					_armorPenetration = value;
```

and similarly below.